vm/io.cpp

```diff
@@ -13,6 +13,39 @@ The Factor library provides platform-specific code for Unix and Windows
 with many more capabilities so these words are not usually used in
 normal operation. */
 
+int raw_fread(void* ptr, size_t size, size_t nitems, FILE* stream) {
+  size_t items_read = 0;
+
+  do {
+    size_t ret = fread((void*)((int*)ptr + items_read * size), size,
+                       nitems - items_read, stream);
+    if (ret == 0) {
```

Comment:
Ok, I'll look into it closer. The idea was to return `-1` on error, therefore the size_t -> int change.